ecosystem/lava-sdk/src/rpcconsumer/rpcconsumer_server.ts

```diff
@@ -140,27 +140,37 @@ export class RPCConsumerServer {
       extraRelayTimeout +
       getTimePerCu(chainMessage.getApi().getComputeUnits()) +
       AverageWorldLatency;
-    try {
-      const consumerSessionsMap = this.consumerSessionManager.getSessions(
-        chainMessage.getApi().getComputeUnits(),
-        unwantedProviders,
-        chainMessage.getRequestedBlock(),
-        "",
-        []
-      );
-      if (consumerSessionsMap instanceof Error) {
-        return consumerSessionsMap;
-      }
-      // TODO: send to several
-      // return this.sendRelayToAllProvidersAndRace(
-      //   consumerSessionsMap,
-      //   extraRelayTimeout
-      // );
-      const firstEntry = consumerSessionsMap.entries().next();
-      if (firstEntry.done) {
-        return new Error("returned empty consumerSessionsMap");
+    const consumerSessionsMap = this.consumerSessionManager.getSessions(
+      chainMessage.getApi().getComputeUnits(),
+      unwantedProviders,
+      LATEST_BLOCK,
+      "",
+      []
+    );
+    if (consumerSessionsMap instanceof Error) {
+      return consumerSessionsMap;
+    }
+
+    if (consumerSessionsMap.size == 0) {
+      return new Error("returned empty consumerSessionsMap");
+    }
+
+    let finalRelayResult: RelayResult | RelayError[] | Error | undefined;
+    let responsesReceived = 0;
+
+    const trySetFinalRelayResult = (
+      res: RelayResult | RelayError[] | Error
```

Comment:
what happens here is that we take only the last reply, this is not the golang functionality in which we take the first reply that is not an error